apis/v1alpha2/validation/httproute_test.go

```diff
@@ -553,3 +558,89 @@ func TestValidateServicePort(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateHTTPRouteTypeMatchesField(t *testing.T) {
+	tests := []struct {
+		name        string
+		routeFilter []gatewayv1a2.HTTPRouteFilter
+		errCount    int
+	}{
+		{
+			name: "valid HTTPRouteFilterRequestHeaderModifier type filter with matching field",
+			routeFilter: []gatewayv1a2.HTTPRouteFilter{{
+				Type:                  gatewayv1a2.HTTPRouteFilterRequestHeaderModifier,
+				RequestHeaderModifier: &gatewayv1a2.HTTPRequestHeaderFilter{},
```

Comment:
I wonder if we need to also make sure that the value of these fields is non empty? Even a simple check like `filter.RequestRedirect == nil || (gatewayv1a2.HTTPRequestHeaderFilter{}) == *filter.RequestRedirect)` might be helpful to ensure completely empty structs were also invalid.

Reply:
Done. I had to use `reflect.DeepEqual` in some cases.